### PR TITLE
Support URL encoded username and password in basic authentication

### PR DESCRIPTION
**Pull request by Choraden**

### CODE REVIEW

**Comment by Choraden** [comment]
By the way, I would like to point out that the forwarder passes the auth header to upstream proxies, while basic auth wrapper for http.Handler deletes it[1]. Is it intentional? 

[1] https://github.com/saucelabs/forwarder/blob/main/middleware/basic_auth.go#L103

File: config.go [hunk 1]
```diff
@@ -33,11 +33,11 @@ func ParseUserInfo(val string) (*url.Userinfo, error) {
 	}
 
 	// URL decode the username and password
-	u, err := url.QueryUnescape(u)
+	u, err := url.PathUnescape(u)
```
Comment (mmatczuk) [inline config.go:36]: Interesting.

File: middleware/basic_auth_test.go [hunk 2]
```diff
@@ -56,3 +56,16 @@ func TestBasicAuthWrap(t *testing.T) {
 		}
 	})
 }
+
+func TestBasicAuthURLEncoded(t *testing.T) {
```
Comment (mmatczuk) [inline middleware/basic_auth_test.go:60]: I do not like that you put things always at the end I causes more conflicts than needed and introduces balagan.
This should live below TestBasicAuth or better be a subtest.

Comment (Choraden) [reply middleware/basic_auth_test.go]: Well, if you think about, it lived below TestBasicAuth, just not directly... It's a subtest now :)

**Comment by mmatczuk** [comment]
Closing this.

**Pull request closed without merging**